### ISSUE TITLE
Location search doesn't find any locations

### ISSUE THREAD

**Issue by nikobockerman** [TylerWilliamson/QuickWeather#48]
# Reproduction steps:
1. Launch QuickWeather
    * this opens the settings for API Key view because there are no locations added yet
    * API Key is already set previously, it's shows green tick

2. Click `>` for next page
    * Locations view opens

3. Click "Other location"
    * "Search for location" view opens

4. Try to search for Berlin
    * Or any other city or country

# Actual result:
"No results" text is shown in red color and nothing is found.

# Expected result:
Some UI element to show search results for Berlin

# More information:
Application version: 2.2.6 from F-Droid
OS info: GrapheneOS based on Android 13

My API key is few months old and it's for OneCall 3.0, in case it matters here. I followed #41 and then got a working api key, but I have never got any search results.
I can add a location to QuickWeather using the map view, but even then I don't get any weather data on main screen. Just a spinning "loading" icon.

I would like to get QuickWeather actually work for me and chose to start resolving the issues from this location search problem. Is there something I can do to get more information of what might be going wrong?

**Comment by TylerWilliamson** [comment]
So there are two things going on here:

1. The search returning no results is due to the GeoCoder not returning any results. The GeoCoder is part of the Android API and uses Google Play Services. You may not have Google Play Services, or your internet traffic to Google's GeoCoder may be blocked (due to a VPN, or your privacy-focused OS GrapheneOS). The solution, as you found, is to use the map or manually enter the coordinates. If you really want to use the search feature, you should check if you have Google Play Services installed and check if there are any features of GrapheneOS which may be blocking the feature. Also, check your internet connection.
2. Not getting weather data on the main screen is a different issue, but may have a similar solution. That means that QuickWeather has not received data from OpenWeatherMap for your selected location. This may be caused by your OS, blocked traffic, or your internet connection.

**Comment by nikobockerman** [comment]
Thanks for the hints!

I found that Geocoder API isn't implemented on GrapheneOS: https://github.com/GrapheneOS/os-issue-tracker/issues/23. So that explains why the location search doesn't get any results.

Not getting any on the main screen is still a problem. I tried to disable VPN, private DNS and everything that I could think that could be blocking based on DNS. And also tried with wifi and mobile data. Those didn't help in this main screen issue. All other apps I have installed can successfully communicate with their necessary services so I'm leaning to think that QuickWeather does something differently that causes this. The `Test API key` in settings does show green so I would guess that it tells that QuickWeather can technically communicate with OpenWeatherMap API server, but then something else goes wrong in the main screen.
Any suggestions what I could look into to understand what goes wrong when the main screen doesn't show anything?

**Comment by nikobockerman** [comment]
I attempted to use MyLocation and with that I got weather data shown. However, I don't want to use that but I'd rather use explicitly added locations. So I wanted to understand why that didn't work.

I ended up learning something about building Android apps, installing them and tracing logs from my phone. And then I built QuickWeather myself, added some more log messages and managed to find what goes wrong and why I don't get anything to see on the screen.

Skipping the intermediate steps, this is the problem: Locale when entering Latitude and Longitude. When I switched my phone to use en-us locale and entered the coordinates in your screenshot, I got the weather data shown in the app. But when I use my default locale fi-fi or my secondary en-fi, the main view doesn't show anything.

I'm guessing the reason to be that `.` is not the decimal separator in Finland but instead `,` is. However, the text boxes for Latitude and Longitude don't allow me to enter `,`, but only `.`. And after I click `OK` and come back to edit the location, the coordinates have changed as if I didn't even enter `.`.

Example:
I enter these: Latitude=`33.749`, Longitude=`-84.388`
=> I click `OK` and come back to edit.
=> These values are in the boxes: Latitude=`33749,000`, Longitude=`-84388,000`

And when I logged the actual request URL it was this:
```
https://api.openweathermap.org/data/2.5/onecall?appid=<appId>&lat=33749.000000&lon=-84388.000000&lang=fi&units=imperial
```

And when I try that with curl, the HTTP 400 response contains this:
```
{
  "cod": "400",
  "message": "wrong latitude"
}
```

**Comment by nikobockerman** [comment]
Adding this property: `android:digits="0123456789,."` in `dialog_editlocation.xml` for both `@+id/editlocation_latitude` and `@+id/editlocation_longitude` allows me to enter comma in those text boxes.

Then I can enter these to the text boxes: Latitude=`33,749`, Longitude=`-84,388`.
And with those values I can see weather data in the main view.

However, I'm not sure if adding those properties is a proper solution or a workaround. They were suggested in some of the first StackOverflow answers I found.

**Comment by nikobockerman** [comment]
For reference, here are plenty of possible ways to deal with this issue: https://stackoverflow.com/questions/3821539/decimal-separator-comma-with-numberdecimal-inputtype-in-edittext

The one I mentioned above using `android:digits="0123456789-,.` is one of them, but it has a downside that it allows multiple separators to be added. But there are other, possibly better ways to deal with this issue which seems to be also an unfixed bug in the Android SDK.

**Comment by TylerWilliamson** [comment]
Interesting. I implemented a fix a while back to accept the `,` separator as well as the `.` separator, and also display it in the Edit menu, but neglected to test that `,` could actually be entered. After some testing, I will include your fix in the next release. I did add `-` to the list (`android:digits="0123456789,.-"`).

**Comment by TylerWilliamson** [comment]
Fixed in [2.2.7](https://github.com/TylerWilliamson/QuickWeather/releases/tag/2.2.7)
